app/controllers/api/v1/web_hooks_controller.rb

```diff
@@ -1,16 +1,17 @@
 class Api::V1::WebHooksController < Api::BaseController
-  before_action :authenticate_with_api_key
+  before_action :find_api_key
```

Comment:
As above, consider using a method name that reveals this performs authentication.